src/engine/source/cmds/src/cmdRun.cpp

```diff
@@ -0,0 +1,163 @@
+#include "cmds/cmdRun.hpp"
+
+#include <atomic>
+#include <csignal>
+#include <thread>
+#include <vector>
+
+#include <hlp/hlp.hpp>
+#include <kvdb/kvdbManager.hpp>
+#include <logging/logging.hpp>
+#include <rxbk/rxFactory.hpp>
+
+#include "base/utils/getExceptionStack.hpp"
+#include "builder.hpp"
+#include "catalog.hpp"
+#include "engineServer.hpp"
+#include "protocolHandler.hpp"
+#include "register.hpp"
+
+namespace
+{
+constexpr auto WAIT_DEQUEUE_TIMEOUT_USEC = 1 * 1000000;
+
+// variables for handling threads
+std::atomic<bool> gs_doRun = true;
+std::vector<std::thread> gs_threadList;
+
+void sigint_handler(const int signum)
+{
+    // Inform threads that they must exit
+    gs_doRun = false;
+
+    for (auto& t : gs_threadList)
+    {
+        t.join();
+    };
+
+    // TODO: this should not be necessary, but server threads are not terminating.
+    exit(0);
+}
+} // namespace
+
+namespace cmd
+{
+void run(const std::string& kvdbPath,
+         const std::string& endpoint,
+         const int queueSize,
+         const int threads,
+         const std::string& fileStorage,
+         const std::string& environment,
+         const int logLevel)
+{
+
+    // Set Crt+C handler
+    sigset_t sig_empty_mask;
+    sigemptyset(&sig_empty_mask);
+
+    struct sigaction sigintAction;
+    sigintAction.sa_handler = sigint_handler;
+    sigintAction.sa_mask = sig_empty_mask;
+
+    sigaction(SIGINT, &sigintAction, NULL);
+
+    // Init logging
+    logging::LoggingConfig logConfig;
+    switch (logLevel)
+    {
+        case 0: logConfig.logLevel = logging::LogLevel::Debug; break;
+        case 1: logConfig.logLevel = logging::LogLevel::Info; break;
+        case 2: logConfig.logLevel = logging::LogLevel::Warn; break;
+        case 3: logConfig.logLevel = logging::LogLevel::Error; break;
+        default: WAZUH_LOG_ERROR("Invalid log level: {}", logLevel);
+    }
+    logging::loggingInit(logConfig);
+
+    KVDBManager::init(kvdbPath);
```

Comment:
there are some duplicated code blocks between test and run that we could move to a separate function:
`Loglevel`, `KVdB init`, `catalog initialization`, etc.
Can this be possible, would it make an easier-to-read code or at least unify where to modify it in the future?

Reply:
Kvdb and catalog initialization are both a call to one function. Making a function to simply call another function seems overkill to me:
```
    KVDBManager::init(kvdbPath);

    catalog::Catalog _catalog(catalog::StorageType::Local, fileStorage);
 ```
 
Same applies to the builder and hlp initialization. Logging could be abstracted, but should be done in the logging module.

Reply:
Right now test and run commands are very similar, but I don't want to couple making abstractions just to save a couple of lines.  When the Engine API is implemented this will change.